components/common/src/templating/mod.rs

```diff
@@ -23,10 +27,33 @@ use handlebars::{Handlebars, RenderError};
 use serde::Serialize;
 use serde_json;
 
-pub use self::context::RenderContext;
 use crate::error::{Error, Result};
+use crate::hcore::fs;
+use crate::hcore::package::PackageInstall;
+use crate::templating::hooks::{Hook, InstallHook};
+
+pub use self::context::RenderContext;
+
+pub fn compile_for_package_install(package: &PackageInstall) -> Result<()> {
+    let pkg = package::Pkg::from_install(package)?;
 
-static LOGKEY: &'static str = "TP";
+    fs::SvcDir::new(&pkg.name, &pkg.svc_user, &pkg.svc_group).create()?;
+
+    let cfg = config::Cfg::new(&pkg, None)?;
+    let ctx = RenderContext::new(&pkg, &cfg);
+    let cfg_renderer = config::CfgRenderer::new(pkg.path.join("config_install"))?;
+    cfg_renderer.compile(&pkg.name, &pkg, &pkg.svc_config_install_path, &ctx)?;
+
+    if let Some(ref hook) = InstallHook::load(
+        &pkg.name,
+        &fs::svc_hooks_path(&pkg.name),
+        &package.installed_path.join("hooks"),
```

Comment:
we don't. You may be thinking of the `hooks` and `config` paths in the `/hab/svc` directory which also has an entry for `config_install`. That said it would not be a bad idea to add them to the `Pkg` struct.